### PR TITLE
feat: add auxiliary tenant ids for service principal

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Adds support for `auxiliaryTenantIDs` for `type: 1` (Service Principals)
  - Only supported with client id and password. No support currently for SP with certificates.
  - The first token is the primary token. Subsequent tokens in the slice are auxiliary tokens. 

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

fixes https://github.com/Azure/aad-pod-identity/issues/814

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Notes for Reviewers**:

### CODE REVIEW

**Comment by aramase** [comment]
This is ready for review. Will add docs for this in the next commit.

File: test/e2e/framework/azureidentity/azureidentity_helpers.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 	"io/ioutil"
 	"strings"
 
+	api "k8s.io/api/core/v1"
+
```
Comment (chewong) [inline test/e2e/framework/azureidentity/azureidentity_helpers.go:13]: nit: import order should be

```
import(
  <standard libs>

  <internal libs>

  <external libs>
)
```

File: test/e2e/framework/azureidentity/azureidentity_helpers.go [hunk 2]
```diff
@@ -9,6 +9,8 @@ import (
 	"io/ioutil"
 	"strings"
 
+	api "k8s.io/api/core/v1"
```
Comment (chewong) [inline test/e2e/framework/azureidentity/azureidentity_helpers.go:12]: naming convention in test/e2e:
```suggestion
	corev1 "k8s.io/api/core/v1"
```

File: pkg/nmi/managed.go [hunk 3]
```diff
@@ -68,7 +68,7 @@ func (mc *ManagedClient) GetIdentities(ctx context.Context, podns, podname, clie
 }
 
 // GetToken returns an ADAL token based on the request and its pod identity.
-func (mc *ManagedClient) GetToken(ctx context.Context, rqClientID, rqResource string, azureID aadpodid.AzureIdentity) (token *adal.Token, err error) {
+func (mc *ManagedClient) GetToken(ctx context.Context, rqClientID, rqResource string, azureID aadpodid.AzureIdentity) (tokens []*adal.Token, err error) {
```
Comment (chewong) [inline pkg/nmi/managed.go:71]: Would renaming this function `GetTokens` make more sense?

File: test/e2e/go.mod [hunk 4]
```diff
@@ -19,4 +19,4 @@ require (
 	sigs.k8s.io/controller-runtime v0.6.0
 )
 
-replace github.com/Azure/aad-pod-identity => ../..
+replace github.com/Azure/aad-pod-identity => ../..
```
Comment (chewong) [inline test/e2e/go.mod:22]: nit: add new line to EOF

File: test/e2e/framework/secret/secret_helpers.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/Azure/aad-pod-identity/test/e2e/framework"
+)
+
+// CreateInput is the input for Create.
+type CreateInput struct {
+	Creator      framework.Creator
+	Config       *framework.Config
+	Name         string
+	Namespace    string
+	ClientSecret string
```
Comment (chewong) [inline test/e2e/framework/secret/secret_helpers.go:24]: we could obtain the SP client secret from `config`, so we don't need an extra field to store `ClientSecret`

**Comment by chewong** [comment]
could you also rebase this PR?

File: test/e2e/framework/azureidentity/azureidentity_helpers.go (hunk 1 above)
Comment (aramase) [reply test/e2e/framework/azureidentity/azureidentity_helpers.go:13]: yeah, Goland IDE does it in a different way. Let me fix it.

File: test/e2e/framework/secret/secret_helpers.go (hunk 5 above)
Comment (aramase) [reply test/e2e/framework/secret/secret_helpers.go:24]: The reason for keeping the `ClientSecret` is to allow each test to decide what they want to pass for `clientSecret`. This'll allow us to add negative test scenarios in the future if any. Lmk what you think. If we want to remove it for now I can make the change.

Comment (chewong) [reply test/e2e/framework/secret/secret_helpers.go:24]: SGTM

**Review by chewong — APPROVED** [review]
Regarding https://github.com/Azure/aad-pod-identity/pull/843#issuecomment-719726819, we could include documentation in a follow-up PR.

File: test/e2e/framework/secret/secret_helpers.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/aad-pod-identity/test/e2e/framework"
```
Comment (chewong) [inline test/e2e/framework/secret/secret_helpers.go:15]: nit: import order

**Pull request merged**